source/guide_prosody.rst

```diff
@@ -0,0 +1,309 @@
+
+TLS
+---
+For simplicity we use openssl and the certificates created via letsencrypt_ for above domains which are already provided on the host after adding the domains and visiting them.
+
+Configure luarocks
+------------------
+
+Prosody is written in ``lua`` and has some dependencies which we install with the package manager ``luarocks``. In order to run the installed packages we have to adapt the path:
+
+::
+
+ PATH=$HOME/.luarocks/bin:$PATH
+ 
+ export $PATH
+
+Additionally we need to provide the paths ``LUA_PATH`` and ``LUA_CPATH`` to use ``luarocks`` accordingly:
+
+::
+
+ [isabell@stardust ~]$ echo "$(luarocks path)" >> ~/.bash_profile \
+ && source ~/.bash_profile
+ [isabell@stardust ~]$
+
+Install lua-dependencies
+------------------------
+
+The following dependencies_ (``luasocket``, ``luaexpat`` and ``luafilesystem``) are required:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luasocket --local && \
+ luarocks install luaexpat --local EXPAT_BINDIR="/usr/bin" EXPAT_INCDIR="/usr/include/" EXPAT_LIBDIR="/usr/lib64" && \
+ luarocks install luafilesystem --local && \
+ luarocks install luasec --local OPENSSL_BINDIR="/usr/bin" OPENSSL_INCDIR="/usr/include" OPENSSL_LIBDIR="/usr/lib64"
+ [...]
+ luasocket [...] is now built and installed in [...]
+ luaexpat [...] is now built and installed in [...]
+ luafilesystem [...] is now built and installed in [...]
+ luasec [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+.. note:: The variables ``*_BINDIR`` ``*__INCDIR`` and ``*_LIBDIR`` are necessary for correct linking the associated library because CentOS uses a different layout for those files than luarocks expects!
+
+Further optional ones (``luadbi-mysql``, ``lua-zlib`` and ``luaevent``) can be installed with these commands:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luadbi-mysql --local MYSQL_BINDIR="/usr/bin" MYSQL_INCDIR="/usr/include/mysql" MYSQL_LIBDIR="/usr/lib64" && \
+ luarocks install lua-zlib --local && \
+ luarocks install luaevent --local
+ [...]
+ luadbi [...] is now built and installed in [...]
+ luadbi-mysql [...] is now built and installed in [...]
+ lua-zlib [...] is now built and installed in [...]
+ luaevent [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+To list the installed packages with their versions use the command ``luarocks list``.
+
+Set c99 compilation flag
+------------------------
+In order to built prosody we have to activate C99-mode as global setting for all packages compiled via luarocks ``~/.luarocks/config-5.1.lua``.
+
+::
+
+ variables = {
+ CC= "gcc -std=c99"
+ }
+
+.. note:: This is a temporary workaround since luarocks not allowing setting this flag via ``.rockspec``-file.
+
+MySQL-database
+--------------
+
+Create a separate db for prosody data:
+
+::
+
+ [isabell@stardust ~]$ mysql -e "CREATE DATABASE ${USER}_prosody CHARACTER SET utf8"
```

Comment:
By default we use `utf8mb4`, which supports emojis and a lot of other characters. The old `utf8` charset can only store 3 bytes, which causes problems. Is there a reason why we need `utf8` here?

Reply:
No specific reason, I was just unaware of this problems.. I removed the definition completely, since other guides don't mention it at all and trusting your default values, right? Note that the seafile guide also uses the utf8 definiton..

Reply:
Seafile doesn't (or didn't) support utf8mb4, see https://github.com/Uberspace/lab/pull/212#discussion_r224017740.